The minimum mode of the key-spacing rule has a bug.

**ESLint version:** the master branch (commit 0acbe57602b1297cc1e7a4b04edac8651ed378f9)

In production code, I use the 1.10.3.

**Lint result:**

```shell
cjc:~/www/node/eslint$ ./bin/eslint.js -c .eslintrc.json index.js 

/home/cjc/www/node/eslint/index.js
  1:1  error  Use the global form of 'use strict'  strict

✖ 1 problem (1 error, 0 warnings)
```

**`eslintrc.json`:**

```js
{
  "rules": {
    "key-spacing": [2, { "mode": "minimum", "beforeColon": false, "afterColon": true }]
  }
}
```

**`index.js`:**

```js
(function() {
    var obj = {
        foo : "Foo"
    };

    obj.foo = "Bar";
}());
```

There should not be an extra space before the colon. I mean an error should be output.


<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30040416-the-minimum-mode-of-the-key-spacing-rule-has-a-bug?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/lib/rules/key-spacing.js
+++ b/lib/rules/key-spacing.js
@@ -203,7 +203,10 @@ module.exports = function(context) {
             firstTokenAfterColon = context.getTokenAfter(getNextColon(key)),
             location = side === "key" ? key.loc.start : firstTokenAfterColon.loc.start;
 
-        if ((diff && mode === "strict" || diff < 0 && mode === "minimum") &&
+        if ((
+            diff && mode === "strict" ||
+            diff < 0 && mode === "minimum" ||
+            diff > 0 && !expected && mode === "minimum") &&
             !(expected && containsLineTerminator(whitespace))
         ) {
             context.report(property[side], location, messages[side], {

--- a/tests/lib/rules/key-spacing.js
+++ b/tests/lib/rules/key-spacing.js
@@ -152,8 +152,7 @@ ruleTester.run("key-spacing", rule, {
     }, {
         code: [
             "obj = { key ",
-            "    :     ",
-            " longName };"
+            "    :longName };"
         ].join("\n"),
         options: [{
             beforeColon: true,
@@ -161,7 +160,7 @@ ruleTester.run("key-spacing", rule, {
             mode: "minimum"
         }]
     }, {
-        code: "obj = { key     :      longName };",
+        code: "obj = { key     :longName };",
         options: [{
             beforeColon: true,
             afterColon: false,
@@ -983,6 +982,23 @@ ruleTester.run("key-spacing", rule, {
         }],
         errors: [
             { message: "Missing space before value for key 'ex'.", line: 4, column: 7, type: "Identifier" }
+        ]
+    }, {
+        code: [
+            "obj = {",
+            "   get fx() { return 'f'; },",
+            "   get gx() { return 'g'; },",
+            "   ex : e",
+            "};"
+        ].join("\n"),
+        options: [{
+            align: "colon",
+            beforeColon: false,
+            afterColon: true,
+            mode: "minimum"
+        }],
+        errors: [
+            { message: "Extra space after key 'ex'.", line: 4, column: 4, type: "Identifier" }
         ]
     }, {
         code: [

